pkg/cmd/server/admission/init.go

```diff
@@ -22,6 +23,7 @@ type PluginInitializer struct {
 	RESTClientConfig      restclient.Config
 	Informers             shared.InformerFactory
 	ClusterQuotaMapper    clusterquotamapping.ClusterQuotaMapper
+	DefaultRegistryFn     imageapi.DefaultRegistryFunc
```

Comment:
The master is the one who needs to know that.  Kubelets should not be allowed to remap pull specs (the whole "add registry with different name" on Docker is not the right design and as far as I'm concerned is unsupported for the platform as a whole).  

Master config should eventually allow alternate names for the registry, and it'll be an array of strings.  Kubelets will not be allowed to have different decisions for what an image registry is.